plaidml/edsl/tests/edsl_test.cc

```diff
@@ -2003,6 +2003,186 @@ TEST_F(CppEdsl, Lens) {
   checkExact(program, {input}, {expected});
 }
 
+TEST_F(CppEdsl, Loop) {
+  auto A = Placeholder(DType::INT32, {4});
+  Tensor O = PLAIDML_EDSL_LOOP(5, A)(Tensor arg) {
+    arg = arg + 1;
+    return arg;
+  };
```

Comment:
Can we make the syntax be like:
```
Tensor O = loop(5, {A}, [&](Tensor iv) {
  return iv + 1;
};
```

Reply:
Yes, I will change to this form. One thing about the macro is that we want to force user to use  ` [=]` instead of ` [&]` in the lambda function, because the C++ portion of the loopbody will be executed once. If a user is using `[&]` and write to c++ variables in the loop body, it probably will not be that the user expected. But as long as the user undestand this, it would be ok.